src/hooks/useAuth.tsx

```diff
@@ -48,12 +53,24 @@ const useAuth = ({ redirectUrl }: { redirectUrl?: string } = {}) => {
   useEffect(() => {
     const pathname = location.pathname as Paths;
 
-    if (loginRequiredPaths.includes(pathname) && !userInfo) {
-      navigate(Paths.home, {
+    if (!isLoading && !userInfo && redirectUrl) {
+      toast(
+        <Text color={colorSet.primary} textAlign={"right"}>
+          공지를 작성하고 리마인드하려면 <br />
+          <Highlight font={Font.Bold}>로그인</Highlight>해주세요!
+        </Text>,
+        {
+          className: "toast-primary-colored",
+          autoClose: false,
+          closeButton: false,
+        },
+      );
+
+      navigate(redirectUrl, {
         state: { from: pathname },
       });
     }
```

Comment:
이거 개인적으로는 return을 먼저하는 것을 선호하는데,
예를 들면

```suggestion
    if (!redirectUrl) return;
    if (isLoading) return;
    if (userInfo) return;
    
    toast(
      <Text color={colorSet.primary} textAlign={"right"}>
        공지를 작성하고 리마인드하려면 <br />
        <Highlight font={Font.Bold}>로그인</Highlight>해주세요!
      </Text>,
      {
        className: "toast-primary-colored",
        autoClose: false,
        closeButton: false,
      },
    );

    navigate(redirectUrl, {
      state: { from: pathname },
    });
```

Reply:
@2paperstar "Commit suggestion"